### ISSUE TITLE
[8.0p1] Frequent hangs when editing code

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#6733]
While editing C# code in 8.0p1 in a relatively small project, I'm seeing frequent hangs of several seconds. If I `killall -QUIT VisualStudio`, the trace is not very helpful. There's nothing much in the UI thread.

Here's a full trace https://gist.github.com/mhutch/02c5133a18665192e7e26b988f39ff5a

A little worryingly, we have *52* FileSystemWatcher threads.

**Comment by se5a** [comment]
Could the number of FileSystemWatcher threads be related to #6726 at all?

**Comment by mhutch** [comment]
Not sure. Have you tried getting a thread dump with `kill -QUIT`?

**Comment by se5a** [comment]
![image](https://user-images.githubusercontent.com/1490538/50733665-81decc80-11f6-11e9-8337-293671f6dbf6.png)

Am I doing it right? nothing happens... MD doesn't quit, and I'm assuming it should output to the term?

**Comment by mhutch** [comment]
Yup the only thread in there that's doing any work is an FSW thread.

**Comment by se5a** [comment]
I'm assuming it's the `"<unnamed thread>"  at System.TimeZoneInfo.GetApplicableRule` thread in my case that's the FSW, only one in mine though?

**Comment by mhutch** [comment]
Yep.

I'm also curious why FSW needs to do timezone conversion...
